### PR TITLE
Added missing public_ip_addresses property to api_management

### PR DESCRIPTION
**Pull request by vidarw**
Adds the missing property fields of *public_ip_addresses* to the terraform api_management resource and data source.

Example code that demonstrates the fixed problem:

```hcl
provider "azurerm" {}

resource "azurerm_resource_group" "test" {
  name     = "amtestRG-vidarw"
  location = "westeurope"
}

resource "azurerm_api_management" "test" {
  name            = "acctestAM-vidarw"
  publisher_name  = "pub1"
  publisher_email = "pub1@email.com"

  sku {
    name     = "Developer"
    capacity = 1
  }

  location            = "${azurerm_resource_group.test.location}"
  resource_group_name = "${azurerm_resource_group.test.name}"
}

data "azurerm_api_management" "test" {
  name                = "${azurerm_api_management.test.name}"
  resource_group_name = "${azurerm_api_management.test.resource_group_name}"
}

resource "local_file" "foo" {
  content = "${data.azurerm_api_management.test.public_ip_addresses.0}"
  filename = "${path.module}/output.txt"
}
```

### CODE REVIEW

File: website/docs/d/api_management.html.markdown [hunk 1]
```diff
@@ -49,6 +49,8 @@ output "api_management_id" {
 
 * `portal_url` - The URL of the Publisher Portal.
 
+* `public_ip_adresses` - The Public IP adresses of the API Management Service.
```
Comment (tombuildsstuff) [inline website/docs/d/api_management.html.markdown]:
```suggestion
* `public_ip_adresses` - The Public IP addresses of the API Management Service.
```

Comment (tombuildsstuff) [inline website/docs/d/api_management.html.markdown]:
```suggestion
* `public_ip_adresses` - The Public IP addresses of the API Management Service.
```

File: website/docs/r/api_management.html.markdown [hunk 2]
```diff
@@ -192,6 +192,8 @@ In addition to all arguments above, the following attributes are exported:
 
 * `portal_url` - The URL for the Publisher Portal associated with this API Management service.
 
+* `public_ip_adresses` - The Public IP adresses of the API Management Service.
```
Comment (tombuildsstuff) [inline website/docs/r/api_management.html.markdown]:
```suggestion
* `public_ip_adresses` - The Public IP addresses of the API Management Service.
```

File: azurerm/resource_arm_api_management_test.go [hunk 3]
```diff
@@ -75,14 +75,15 @@ func TestAccAzureRMApiManagement_complete(t *testing.T) {
 				Check: resource.ComposeTestCheckFunc(
 					testCheckAzureRMApiManagementExists(resourceName),
 					resource.TestCheckResourceAttr(resourceName, "tags.Acceptance", "Test"),
+					resource.TestCheckResourceAttrSet(resourceName, "public_ip_addresses"),
```
Comment (tombuildsstuff) [inline azurerm/resource_arm_api_management_test.go]: could we ensure there's items in this list? we can do that via 

```suggestion
					resource.TestCheckResourceAttrSet(resourceName, "public_ip_addresses.#"),
```

File: azurerm/data_source_api_management_test.go [hunk 4]
```diff
@@ -26,6 +26,7 @@ func TestAccDataSourceAzureRMApiManagement_basic(t *testing.T) {
 					resource.TestCheckResourceAttr(dataSourceName, "sku.0.capacity", "1"),
 					resource.TestCheckResourceAttr(dataSourceName, "sku.0.name", "Developer"),
 					resource.TestCheckResourceAttr(dataSourceName, "tags.%", "0"),
+					resource.TestCheckResourceAttrSet(dataSourceName, "public_ip_addresses"),
```
Comment (tombuildsstuff) [inline azurerm/data_source_api_management_test.go]: can we ensure there's items in this list? we can do that via:

```suggestion
					resource.TestCheckResourceAttrSet(dataSourceName, "public_ip_addresses.#"),
```

File: website/docs/d/api_management.html.markdown (hunk 1 above)
Comment (vidarw) [reply website/docs/d/api_management.html.markdown]: Corrected the mistake in the latest commit

Comment (vidarw) [reply website/docs/d/api_management.html.markdown]: Corrected the mistake in the latest commit

File: website/docs/r/api_management.html.markdown (hunk 2 above)
Comment (vidarw) [reply website/docs/r/api_management.html.markdown]: Corrected the mistake in the latest commit

File: azurerm/data_source_api_management_test.go (hunk 4 above)
Comment (vidarw) [reply azurerm/data_source_api_management_test.go]: Improved the test. Thanks for the suggestion.

**Comment by tombuildsstuff** [comment]
Tests pass:

screenshot 2018-11-14 at 17 17 57

**Pull request merged**